airflow/models/dagrun.py

```diff
@@ -173,7 +173,12 @@ class DagRun(Base, LoggingMixin):
         uselist=False,
         viewonly=True,
     )
-    dag_run_note = relationship("DagRunNote", back_populates="dag_run", uselist=False)
+    dag_run_note = relationship(
+        "DagRunNote",
+        cascade="save-update, merge, delete, delete-orphan",
```

Comment:
```suggestion
        cascade="all, delete-orphan",
```
`all` means `save-update, merge, refresh-expire, expunge, delete`. I think that's what we need